src/test/kotlin/org/rust/lang/core/type/RsFunctorTypeInferenceTest.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.type
+
+import org.intellij.lang.annotations.Language
+import org.rust.lang.core.psi.RsBlock
+import org.rust.lang.core.types.type
+
+
+class RsFunctorTypeInferenceTest: RsTypificationTestBase() {
+    fun `test if with all branches diverging`() = testFunctor("""
+        fn main() {
+            let x = if true { return; } else { return; };
+            x;
+        }
+      //^ ()
```

Comment:
Previously this tests all checked the type of x.
Is this change really indented to check the type of the fn?